Framework/Geometry/inc/MantidGeometry/Objects/MeshObject.h

```diff
@@ -0,0 +1,212 @@
+#ifndef MANTID_GEOMETRY_MESHOBJECT_H_
+#define MANTID_GEOMETRY_MESHOBJECT_H_
+
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+#include "MantidGeometry/DllConfig.h"
+#include "MantidGeometry/Objects/IObject.h"
+
+#include "BoundingBox.h"
+#include <map>
+#include <memory>
+
+namespace Mantid {
+//----------------------------------------------------------------------
+// Forward declarations
+//----------------------------------------------------------------------
+namespace Kernel {
+class PseudoRandomNumberGenerator;
+class Material;
+class V3D;
+}
+
+namespace Geometry {
+class CacheGeometryHandler;
+class CompGrp;
+class GeometryHandler;
+class Track;
+class vtkGeometryCacheReader;
+class vtkGeometryCacheWriter;
+
+/**
+\class MeshObject
+\brief Triangular Mesh Object
+\version 1.0
+\date Dec 2017
+\author S. Ansell
+
+Mesh Object of Traingles
+
+Copyright &copy; 2017-2018 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class MANTID_GEOMETRY_DLL MeshObject : public IObject {
+public:
+  /// Default constructor
+  MeshObject();
+  /// Constructor from string.
+  MeshObject(const std::string &shapeXML);
+  /// Copy constructor
+  MeshObject(const MeshObject &);
+  /// Assignment operator
+  MeshObject &operator=(const MeshObject &);
+  /// Destructor
+  virtual ~MeshObject();
+  /// Clone
+  IObject *clone() const override {
+    if (m_vertices.size() > 0)
+      return new MeshObject(*this);
+    else {
+      throw("Unintialised MeshObject cannot be copied");
```

Comment:
As i mention in the implementation (cpp file) Would be much better if an uninitialized implementation was not possible.